src/main/java/org/baldogru/imagegallery/factory/ValidationSummaryFactoryImpl.java

```diff
@@ -1,19 +1,41 @@
 package org.baldogru.imagegallery.factory;
 
+import org.baldogru.imagegallery.constants.ValidationStatus;
 import org.baldogru.imagegallery.model.dto.ValidationResult;
 import org.baldogru.imagegallery.model.dto.ValidationSummary;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
 @Service
 class ValidationSummaryFactoryImpl implements ValidationSummaryFactory {
 
-    ValidationSummaryFactoryImpl() {}
+    ValidationSummaryFactoryImpl() {
+    }
 
     @Override
     public ValidationSummary create(List<ValidationResult> validationResults) {
-        //Smialo dawaj jakas swoja implementacje tworzenia tego ValidationSummary
-        return null;
+        List<ValidationResult> successValidationResults =
```

Comment:
Tu dwie sprawy:

1. Jak korzystasz ze streamów to każda operacja w oddzielnej linijce -> chodzi o czytelność tego kodu, a jeszcze najnowszy intelliJ ma ficzera, że pokazuje w locie co dana operacja streama zwraca.
2. Dlaczego filtrujesz tylko te wyniki co miały sukces? Wszystkie powinny trafić do podsumowania.

Reply:
2. Tutaj filtruje tylko success w celu sprawdzenia czy wszystkie przeszły validacje. Bo jeśli tak, to ustawiam status ValidationStatus = SUCCESS, jesli choć jeden nie przeszedł to daje na fail. 
A do listy <ValidationResult> wsadzam tą listę którą otrzymałem w parametrze metody create. 
Myślałem, że na tej zasadzie to ma działać? :P